features/woopit.feature

```diff
@@ -522,3 +522,300 @@ Feature: Woopit
+          "taxIncludedAmount":499,
+          "taxAmount":83,
+          "currency":"EUR"
+        },
+        "vehicleType":"VEHICLE_TYPE_BIKE"
+      }
+      """
+    When I add "Content-Type" header equal to "application/json"
+    And I add "Accept" header equal to "application/json"
+    # https://woop.stoplight.io/docs/carrier/b3A6MzYyMDcwOTU-delivery-request
+    And the store with name "Acme" sends a "POST" request to "/api/woopit/deliveries" with body:
+      """
+      {
+        "orderId": "65zq1d5qz1d56q1",
+        "referenceNumber": "65zq1d5qz1d56q1-456",
+        "quoteId": "yXJPEP30xgja",
+        "picking": {
+          "location": {
+            "addressLine1": "24, Rue de la Paix",
+            "addressLine2": "",
+            "postalCode": "75002",
+            "city": "Paris",
+            "country": "FR"
+          },
+          "infos": "Travaux sur le boulevard.",
+          "interval": [
+            {
+              "start": "2019-12-04T12:30:00+0000",
+              "end": "2019-12-04T13:30:00+0000"
+            }
+          ],
+          "contact": {
+            "firstName": "joe",
+            "lastName": "doe",
+            "phone": "+3312345678"
+          }
+        },
+        "delivery": {
+          "location": {
+            "addressLine1": "48, Rue de Rivoli",
+            "addressLine2": "",
+            "elevator": true,
+            "floor": 2,
+            "postalCode": "75004",
+            "city": "Paris",
+            "country": "FR"
+          },
+          "interval": [
+            {
+              "start": "2019-12-06T13:00:00+0000",
+              "end": "2019-12-06T15:00:00+0000"
+            }
+          ],
+          "contact": {
+            "firstName": "path",
+            "lastName": "met",
+            "phone": "+3312345687"
+          }
+        },
+        "packages": [
+          {
+            "id": 1,
+            "length": {
+              "value": 121.8,
+              "unit": "cm"
+            },
+            "width": {
+              "value": 77.5,
+              "unit": "cm"
+            },
+            "height": {
+              "value": 55,
+              "unit": "cm"
+            },
+            "weight": {
+              "value": 33.4,
+              "unit": "kg"
+            },
+            "products": [
+              {
+                "type": "TYPOLOGY_GENERIC",
+                "ean": "4897567123548",
+                "cug": "e21fe21",
+                "label": "Produit label 1",
+                "quantity": 1
+              }
+            ],
+            "quantity": 2
+          }
+        ],
+        "services": [
+          "SERVICE_SELECTED_ROOM"
+        ]
+      }
+      """
+    Then the response status code should be 200
+    And the response should be in JSON
+    And the JSON should match:
+      """
+      {
+        "deliveryId":"yXJPEP30xgja"
+      }
+      """
+    When I add "Content-Type" header equal to "application/merge-patch+json"
```

Comment:
Does it work without this header? Not sure it will be sent by Woop, I think they will send `Content-Type: application/json` as per the documentation

Reply:
Maybe you will need to use this? https://api-platform.com/docs/core/content-negotiation/#configuring-formats-for-a-specific-resource-or-operation